vss-processor/src/main/kotlin/org/eclipse/kuksa/vssprocessor/spec/VssNodeSpecModel.kt

```diff
@@ -186,31 +207,62 @@ internal class VssNodeSpecModel(
             .build()
     }
 
-    private fun createValueSpec(): Pair<PropertySpec, ParameterSpec> {
-        val valuePropertySpec = PropertySpec
-            .builder(PROPERTY_VALUE_NAME, datatypeProperty)
-            .initializer(PROPERTY_VALUE_NAME)
+    private fun createVssSignalSpec(): Triple<ParameterizedTypeName, MutableList<PropertySpec>, ParameterSpec?> {
+        val propertySpecs = mutableListOf<PropertySpec>()
+        var parameterSpec: ParameterSpec? = null
+
+        val vssSignalMembers = VssSignal::class.declaredMemberProperties
+        vssSignalMembers.forEach { member ->
+            val memberName = member.name
+            when (val memberType = member.returnType.asTypeName()) {
+                genericClassTypeName -> {
+                    val genericClassSpec = createGenericClassSpec(
+                        memberName,
+                        memberType,
+                        datatypeTypeName.toString(),
+                    )
+                    propertySpecs.add(genericClassSpec)
+                }
+
+                else -> {
+                    val (classPropertySpec, classParameterSpec) = createClassParamSpec(
+                        memberName,
+                        valueTypeName,
+                        defaultValue,
+                    )
+                    parameterSpec = classParameterSpec
+                    propertySpecs.add(classPropertySpec)
+                }
+            }
+        }
```

Comment:
Found this, while I wanted to introduce a new property "unit" to VssSignal, which is outputted incorrectly as an int instead of a String. 

In line 215 you iterate over all members of vssSignalClass. Right now we only have dataType and value. 

While dataType is a "genericClassTypeName" value will run into the else branch and execute "createClassParamSpec". 

Right now a strong coupling between the property value and createClassParamSpec exist, because valueTypeName and defaultValue are coupled to the dataType of value and not the field itself. This makes the forEach kinda useless :D 

Just highlighting this for completeness, I can fix this as part of #95

Reply:
Sounds good to me to fix it in #95 where the new property is introduced :)